apis/python/src/tiledbsoma/_soma_object.py

```diff
@@ -333,5 +333,9 @@ def _check_open_read(self) -> None:
         if self.mode != "r":
             raise ValueError(f"{self} is open for writing, not reading")
 
+    def _check_open_write(self) -> None:
+        if self.mode != "w":
+            raise ValueError(f"{self} is open for reading, not writing")
```

Comment:
it also could be closed, right?  So I suggest the error message be:   `f"{self} - must be open for write."` (or something like that)

Reply:
@bkmartinjr you are correct

I copy-pasted from `_check_open_read` which shares this defect.

On thisPR, I'll fix them both.

Or, better, I'll put up a separate PR fixing `_check_open_read`, and rebase this on top of that.

Reply:
> Or, better, I'll put up a separate PR fixing _check_open_read, and rebase this on top of that.

@bkmartinjr that PR is #3847. I've not yet rebased this PR on top of that one, but I will do so today.

I'll put this PR back in draft status until it is once again ready for review (at present, it is not ready for re-review).